### PR TITLE
MGMT-16283: Adding support for caBundle config and icsp

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: internal/clusterconfig/clusterconfig.go [hunk 1]
```diff
@@ -40,7 +42,11 @@ const (
 
 	clusterIDFileName = "cluster-id-override.json"
 
-	idmsFileName = "image-digest-mirror-set.json"
+	idmsFileName  = "image-digest-mirror-set.json"
+	icspsFileName = "image-content-policy-list.json"
+
+	caBundleCMName   = "user-ca-bundle"
```
Comment (browsell) [inline internal/clusterconfig/clusterconfig.go:48]: This assumes the configmap name is user-ca-bundle ? Not sure we can assume that

Comment (tsorya) [reply internal/clusterconfig/clusterconfig.go:48]: Actually we can. This is hardcoded name in mco too

File: internal/clusterconfig/clusterconfig.go [hunk 2]
```diff
@@ -295,3 +307,64 @@ func (r *UpgradeClusterConfigGather) getIDMSs(ctx context.Context) (v1.ImageDige
...
+	r.Log.Info("Fetching user ca bundle")
+	caBundle := &corev1.ConfigMap{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: caBundleCMName,
+		Namespace: configNamespace}, caBundle)
+	if err != nil && errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("failed to get ca bundle cm, err: %w", err)
+	}
```
Comment (sakhoury) [inline internal/clusterconfig/clusterconfig.go:356]: Perhaps we can replace these lines with the function `GetConfigMap` in [func GetConfigMap(ctx context.Context, c client.Client, configMap v1alpha1.ConfigMapRef) (*corev1.ConfigMap, error) {](https://github.com/openshift-kni/lifecycle-agent/blob/main/internal/common/utils.go#L32-L43).

Comment (tsorya) [reply internal/clusterconfig/clusterconfig.go:356]: I can, just not fully understand why we actually have this util function and what it gives us. Will be glad to understand.

Comment (sakhoury) [reply internal/clusterconfig/clusterconfig.go:367]: Sure! This function, was defined and, is used by several plugins. The intent is to define commonly used functions in a unified location that can be easily referenced. In your case, you only need to fetch a ConfigMap once, however, there are cases where ConfigMaps need to be retrieved in several places - thus the reasoning behind its existence (avoid code duplication).

**Pull request merged**